cli/command/manifest/util.go

```diff
@@ -76,6 +76,8 @@ func getManifest(ctx context.Context, dockerCli command.Cli, listRef, namedRef r
 		return dockerCli.RegistryClient(insecure).GetManifest(ctx, namedRef)
 	case err != nil:
 		return types.ImageManifest{}, err
+	case len(data.Raw) == 0:
+		return dockerCli.RegistryClient(insecure).GetManifest(ctx, namedRef)
```

Comment:
In https://github.com/docker/cli/blob/ebaaefe2af5023915793791d3cb6bd305ec9478b/cli/command/manifest/util.go#L75 we could just have:

```suggestion
	case store.IsNotFound(err), len(data.Raw) == 0:
		return dockerCli.RegistryClient(insecure).GetManifest(ctx, namedRef)
```

Reply:
`len(data.Raw) == 0` will always be true if there is an error, so we need to make sure `err != nil` first.